apps/fxc-front/src/app/logic/score/improvedScorer.ts

```diff
@@ -0,0 +1,231 @@
+import { Point } from '@flyxc/common';
+import { Point as XcScorePoint, scoringRules, Solution } from 'igc-xc-score';
+import { BRecord, IGCFile } from 'igc-parser';
+import { CircuitType, Score } from './scorer';
+import { LeagueCode } from './league';
+
+async function lazyLoadedSolver(): Promise<typeof _solver> {
+  if (!_solver) {
+    const { solver } = await import('igc-xc-score');
+    _solver = solver;
+  }
+  return _solver;
+}
+
+let _solver: (
+  flight: IGCFile,
+  scoringRules: object,
+  config?: { [key: string]: any } | undefined,
+) => Iterator<Solution, Solution>;
+
+export type ScoreAndRoute = { score: Score; route: Point[] };
+export type ScoringTrack = { lat: number[]; lon: number[]; timeSec: number[]; minTimeSec: number };
+
+export async function scoreTrack(track: ScoringTrack, leagueCode: LeagueCode): Promise<ScoreAndRoute | undefined> {
```

Comment:
question: should we consider a way to abort the scoring? i.e. passing an `abortController`

Reply:
I am not sure that it is mandatory for the moment.
May be we should wait for user feedbacks before introducing this feature.
A fist step could be a spinner like display when computation is two slow.
Do we have a way to gather computation durations for analysis?